src/node/config.ts

```diff
@@ -33,6 +33,7 @@ export interface UserConfig<ThemeConfig = any> {
   base?: string
   title?: string
   description?: string
+  socialImage?: string
```

Comment:
I don't think we need a site-wide config option for this, since you can already do that using `head`.